vignettes/cpp11_faq.Rmd

```diff
@@ -0,0 +1,184 @@
+---
+title: "cpp11 FAQ"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{cpp11 FAQ}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+Intended to answer common questions about the usage of cpp11 objects
+
+## FAQ
+
+* How do I modify a vector in place?
+
+  * Using `writable::` will always make a copy, but it has a move constructor, so you can use `cpp11::writable::integers(std::move(x))` and it won't make a copy of the data.
+
+* How do I add elements to a named list?
```

Comment:
I think we should explain what we are doing first, then show the code, not just showing code.

Maybe something like

> Use the vector's `push_back()` method along with named literals.

In the code examples you can use the cpp11 knitr engine, you can do this with the following header.

    ```{cpp11}

For example https://github.com/r-lib/cpp11/blob/4ef13b7ecdcb9cbd058456b7a18c1bb3ba665d05/vignettes/motivations.Rmd#L134-L144